internal/postgres/process.go

```diff
@@ -31,11 +31,14 @@ import (
 // In schema mode, ProcessPgDump incrementally builds a schema (updating conv).
 // In data mode, ProcessPgDump uses this schema to convert PostgreSQL data
 // and writes it to Spanner, using the data sink specified in conv.
-func ProcessPgDump(conv *Conv, r *internal.Reader) {
+func ProcessPgDump(conv *Conv, r *internal.Reader) error {
 	for {
 		startLine := r.LineNumber
 		startOffset := r.Offset
-		b, stmts := readAndParseChunk(conv, r)
+		b, stmts, err := readAndParseChunk(conv, r)
+		if err != nil {
```

Comment:
This change means that if there is any error during parsing, we'd return and immediately exit (with the changes in main.go). In other words, we'd drop any chunks of pg_dump data that we'd successfully parsed.

It might be better to proceed with processing partial data (since that might be useful), and only exit if we didn't parse anything (i.e. we have the empty schema, which is basically useless).

Reply:
According to `readAndParseChunk`, if an error happens, that means it has already tried to parse the content from where the error happens up to the end of the file, but failed to do so. 

From my understanding, the workflow looks like this: 

```bash

=== first pass begins ===

sql file: 
valid statement #1 (successfully parsed)
invalid statement #2 (error happens)
valid statement #3 (unable to parse this statement)

=== first pass ends ===

=== generate schema file ===

=== create database ===

=== second pass begins === (read data statements and insert/update rows)

=== second pass ends ===

```

In this case, the schema file is not empty, which has the valid statement \# 1 but not \# 2 and \# 3. Do we still need to create the database? and how about the second pass (read data and insert/update rows)?

Please forgive me if I misunderstood some info.

Reply:
Yes, that's correct. We'll process up to the first error, and ignore the rest of the input, both schema and data.

My original thinking was that we should try to make as much progress as possible i.e. that even partial conversion was still useful. But maybe that's not worth doing, since there is a potential cost of increased complexity of behaviour: if you hit a parse error, and some tables were missing, wouldn't that be confusing? Perhaps its simpler to just unambiguously fail. What do you think?

Reply:
I agree that it's simpler to fail quickly. An incomplete schema does not make sense to continue the migration. 

Still use my previous example: 

```sql
CREATE TABLE public.testusers (
    id integer NOT NULL,
    active boolean,
    name text
);

ALTER TABLE public.testusers OWNER TO postgres;

# Error happens here!!!
COPY public.testusers (id, active, name) FROM stdin;
1	t	foo
2	f	bar
\.

ALTER TABLE ONLY public.testusers
    ADD CONSTRAINT testusers_pkey PRIMARY KEY (id);
```

In the above example, if we only have `CREATE TABLE` (the first statement) in our schema file without `ALTER TABLE` (the last one), this schema file is invalid. The table we create would be wrong without a primary key. In this case, we should not try to create the database and insert data.